packages/nimble-components/src/table-column/mixins/sortable-column.ts

```diff
@@ -0,0 +1,70 @@
+import { attr, nullableNumberConverter } from '@microsoft/fast-element';
+import type { TableColumn } from '../base';
+import { TableColumnSortDirection } from '../../table/types';
+
+// Pick just the relevant properties the mixin depends on (typescript complains if the mixin declares private / protected base exports)
+type SortableTableColumn = Pick<TableColumn, 'columnInternals'>;
+// prettier-ignore
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+type SortableTableColumnConstructor = abstract new (...args: any[]) => SortableTableColumn;
+
+// As the returned class is internal to the function, we can't write a signature that uses is directly, so rely on inference
+// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types, @typescript-eslint/explicit-function-return-type
+export function mixinSortableColumnAPI<
+    TBase extends SortableTableColumnConstructor
+>(base: TBase) {
+    /**
+     * The Mixin that provides a concrete column with the API to allow sorting
+     * by the values in that column.
+     */
+    abstract class SortableColumn extends base {
+        public sortingDisabled = false;
+
+        public sortIndex?: number | null = null;
```

Comment:
Can we leave this uninitialized to undefined like it was previously in base? (prefer undefined over null)

```suggestion
        public sortIndex?: number | null;
```

Reply:
Hopefully we can do the same for groupIndex above as well